Better exception messages when statements fail

DouglasHeriot opened the pull request:
Added new constructor to Exception that takes sqlite3* pointer, allowing for getting more information out (sqlite3_errmsg). Makes it much easier to debug syntax errors in prepared statements, for example.

Also implements suggestion from #62 (storing error code).

SRombauts commented:
Hi!

Thanks a lot for this work, seems great at first glance!

A question first: why using both sqlite3_errmsg() and  sqlite3_errstr(), my understanding is that the second one (the more recent one) superseded the old one: does errmsg() sometimes provide more information than the first one?

And can you please fix the compile errors?
 - https://travis-ci.org/SRombauts/SQLiteCpp/jobs/132862602
 - https://travis-ci.org/SRombauts/SQLiteCpp/jobs/132862601

Cheers!

SRombauts commented:
Also, I think public accessors to private variable members would better fit the code used in the rest of the library.

DouglasHeriot commented:
I think I’ve fixed everything now:

* Oops, I only tested against my own app, which was using C++11. Simplified that code now, and tested properly.
* Added public accessors
* I think you’re right that `sqlite3_errmsg` supersedes `sqlite3_errstr`, sometimes `errmsg` is the same as `errstr`, but often it has much more detail (ie. syntax errors) – so, now it only uses `errmsg` (but there's an accessor for `errstr` if needed)
* Used these new `Exception` constructors everywhere appropriate

SRombauts commented:
Thanks a lot for your great contribution, very appreciated!
